### ISSUE TITLE
LibCurlHttpContentFetcher::getContent always returns nullptr

### ISSUE THREAD

**Issue by jbfitbit** [alexa/avs-device-sdk#1364]
## Briefly summarize your issue:
[LibCurlHttpContentFetcher::getContent(...)
](https://github.com/alexa/avs-device-sdk/blob/c93a898b76b02c591daffee832f5f7cd64c46205/AVSCommon/Utils/src/LibcurlUtils/LibCurlHttpContentFetcher.cpp#L268) Always returns a nullptr

## What is the expected behavior?
I'd expect it to return a `std::unique_ptr<avsCommon::utils::HTTPContent>` since that's what the method signature says. If not, I'd expect the method signature to change to `void` so that clients of the SDK can update accordingly.

## What behavior are you observing?
All calls to `LibCurlHttpContentFetcher::getContent(...)` return a `nullptr`. **There is no return statement in that function that _does not_ return `nullptr`.**

## Provide the steps to reproduce the issue, if applicable:
Check out the code

**Comment by dhpp** [comment]
Thanks @jbfitbit for pointing this out.  I suspect the code was updated in a recent version and drifted from the documentation.  I've created an internal tracking ticket for us to address - we'll likely update the documentation for this specialization so that the base class avsCommon::sdkInterfaces::HTTPContentFetcherInterface still allows this functionality if desired.  Thanks again - I'll close out this ticket in the meantime.
